modules/autotagging-commons/src/main/java/org/opensearch/rule/rest/RestCreateRuleAction.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.rule.rest;
+
+import org.opensearch.core.rest.RestStatus;
+import org.opensearch.core.xcontent.ToXContent;
+import org.opensearch.core.xcontent.XContentParser;
+import org.opensearch.rest.BaseRestHandler;
+import org.opensearch.rest.BytesRestResponse;
+import org.opensearch.rest.RestChannel;
+import org.opensearch.rest.RestHandler;
+import org.opensearch.rest.RestRequest;
+import org.opensearch.rest.RestResponse;
+import org.opensearch.rest.action.RestResponseListener;
+import org.opensearch.rule.CreateRuleRequest;
+import org.opensearch.rule.CreateRuleResponse;
+import org.opensearch.rule.action.CreateRuleAction;
+import org.opensearch.rule.autotagging.FeatureType;
+import org.opensearch.rule.autotagging.Rule.Builder;
+import org.opensearch.transport.client.node.NodeClient;
+import org.joda.time.Instant;
+
+import java.io.IOException;
+import java.util.List;
+
+import static org.opensearch.rest.RestRequest.Method.POST;
+import static org.opensearch.rest.RestRequest.Method.PUT;
+import static org.opensearch.rule.rest.RestGetRuleAction.FEATURE_TYPE;
+
+/**
+ * Rest action to create a Rule
+ * @opensearch.experimental
+ */
+public class RestCreateRuleAction extends BaseRestHandler {
+    /**
+     * constructor for RestCreateRuleAction
+     */
+    public RestCreateRuleAction() {}
+
+    @Override
+    public String getName() {
+        return "create_rule";
+    }
+
+    @Override
+    public List<Route> routes() {
+        return List.of(new RestHandler.Route(POST, "_rules/{featureType}/"), new RestHandler.Route(PUT, "_rules/{featureType}/"));
```

Comment:
I believe only `PUT` should be used for creation. Also, trailing / can be removed